modules/kubernetes/ingress-healthz/templates/values.yaml.tpl

```diff
@@ -28,11 +28,21 @@ ingress:
   hostname: ingress-healthz.${dns_zone}
   annotations:
     cert-manager.io/cluster-issuer: letsencrypt
+    %{ if linkerd_enabled }
+    nginx.ingress.kubernetes.io/configuration-snippet: |
+      proxy_set_header l5d-dst-override $service_name.$namespace.svc.cluster.local:$service_port;
+      grpc_set_header l5d-dst-override $service_name.$namespace.svc.cluster.local:$service_port;
```

Comment:
So do all ingress resources need this when linkerd is enabled? Even when they are not using the sidecar? Could be solve with with a mutation?

Reply:
These two headers are always needed if proxy injection is enabled. My opinion as of right now is that I don't want to hide away this complexity from the developer but instead make them understand it. If it becomes a problem, perhaps we can create a mutation for it - but another PR.

Do you agree?

Reply:
I've also updated the README with more info regarding this.

Reply:
Sounds good.